File: mobile/src/api/context.ts
export type InputTask = {
  id: string;
  title: string;
  status: "planned" | "completed" | "cancelled";
  assigned_date: string; // YYYY-MM-DD
  category?: string | null;
  priority?: number | null;
  defer_count?: number;
};

export type FocusItem = {
  task_id: string;
  title: string;
  reason: string;
  score: number;
};

export type OutstandingItem = {
  task_id: string;
  title: string;
  note: string;
};

export type NewsItem = {
  title: string;
  source: string;
  url: string;
  relevance_note: string;
};

export type ContextTodayResponse = {
  date: string;
  summary_line: string;
  focus: FocusItem[];
  outstanding: OutstandingItem[];
  news: NewsItem[];
  nudge?: string | null;
};

const API_BASE = "http://127.0.0.1:8000/api";

export async function fetchTodayContext(tasks: InputTask[]): Promise<ContextTodayResponse> {
  const res = await fetch(`${API_BASE}/context/today`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      tasks,
      include_news: true,
      max_focus: 2,
      max_outstanding: 1,
      max_news: 2,
    }),
  });

  if (!res.ok) {
    throw new Error(`Context request failed: ${res.status}`);
  }

  return res.json();
}

File: backend/app/models/context.py
from pydantic import BaseModel
from typing import Literal, Optional


TaskStatusLiteral = Literal["planned", "completed", "cancelled"]


class InputTask(BaseModel):
    id: str
    title: str
    status: TaskStatusLiteral
    assigned_date: str
    category: Optional[str] = None
    priority: Optional[int] = None
    defer_count: int = 0


class ContextTodayRequest(BaseModel):
    date: Optional[str] = None
    tasks: list[InputTask]
    include_news: bool = False
    max_focus: int = 3
    max_outstanding: int = 3
    max_news: int = 0


class FocusItem(BaseModel):
    task_id: str
    title: str
    reason: str
    score: float


class OutstandingItem(BaseModel):
    task_id: str
    title: str
    note: str


class NewsItem(BaseModel):
    title: str
    source: str
    url: str
    relevance_note: str


class ContextTodayResponse(BaseModel):
    date: str
    summary_line: str
    focus: list[FocusItem]
    outstanding: list[OutstandingItem]
    news: list[NewsItem]
    nudge: Optional[str] = None